File: refactor.js
const fs = require('fs');
const path = require('path');

const dirPath = 'd:/purelab';

function getFiles(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            if (!file.includes('node_modules') && !file.includes('.next') && !file.includes('.git')) {
                results = results.concat(getFiles(file));
            }
        } else {
            if (file.endsWith('.tsx') && !file.endsWith('GlassIcon.tsx')) {
                results.push(file);
            }
        }
    });
    return results;
}

const files = getFiles(dirPath);
const importRegex = /import\s+\{([^}]+)\}\s+from\s+["']lucide-react["'];?/;

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    const match = content.match(importRegex);
    
    if (match) {
        const importsStr = match[1];
        const icons = importsStr.split(',').map(i => i.trim()).filter(i => i);
        
        let newContent = content;
        icons.forEach(icon => {
            let iconName = icon;
            if (icon.includes(' as ')) {
                iconName = icon.split(' as ')[1].trim();
            }
            
            // Replace self closing
            const tagRegex = new RegExp(`<\\s*${iconName}\\b([^>]*)/>`, 'g');
            newContent = newContent.replace(tagRegex, `<GlassIcon icon={${iconName}} $1/>`);
            
            // Replace open/close
            const tagRegexOpen = new RegExp(`<\\s*${iconName}\\b([^>]*)>(.*?)</\\s*${iconName}\\s*>`, 'gs');
            newContent = newContent.replace(tagRegexOpen, `<GlassIcon icon={${iconName}} $1/>`);
        });
        
        if (!newContent.includes('GlassIcon')) {
            newContent = newContent.replace(match[0], match[0] + '\nimport { GlassIcon } from "@/components/ui/GlassIcon";');
        }
        
        if (newContent !== content) {
            fs.writeFileSync(file, newContent, 'utf8');
            console.log(`Updated ${file}`);
        }
    }
});

File: refactor.py
import os
import re

dir_path = "d:/purelab"

def get_files(d):
    files = []
    for root, dirs, filenames in os.walk(d):
        if 'node_modules' in root or '.next' in root or '.git' in root:
            continue
        for f in filenames:
            if f.endswith('.tsx') and f != 'GlassIcon.tsx':
                files.append(os.path.join(root, f))
    return files

files = get_files(dir_path)

import_regex = re.compile(r'import\s+\{([^}]+)\}\s+from\s+["\']lucide-react["\'];?')

for f in files:
    with open(f, 'r', encoding='utf-8') as file:
        content = file.read()
    
    match = import_regex.search(content)
    if match:
        imports_str = match.group(1)
        icons = [i.strip() for i in imports_str.split(',')]
        
        new_content = content
        for icon in icons:
            if not icon: continue
            if ' as ' in icon:
                icon_name = icon.split(' as ')[1].strip()
            else:
                icon_name = icon
                
            tag_regex = re.compile(r'<\s*' + icon_name + r'\b([^>]*)/>')
            new_content = tag_regex.sub(r'<GlassIcon icon={' + icon_name + r'} \1/>', new_content)
            
            tag_regex_open = re.compile(r'<\s*' + icon_name + r'\b([^>]*)>(.*?)</\s*' + icon_name + r'\s*>', re.DOTALL)
            new_content = tag_regex_open.sub(r'<GlassIcon icon={' + icon_name + r'} \1/>', new_content)
        
        if 'GlassIcon' not in new_content:
            new_content = new_content.replace(match.group(0), match.group(0) + '\nimport { GlassIcon } from "@/components/ui/GlassIcon";')
            
        if new_content != content:
            with open(f, 'w', encoding='utf-8') as file:
                file.write(new_content)
                print(f"Updated {f}")
